src/client/Client.cc

```diff
@@ -1709,14 +1709,18 @@ mds_rank_t Client::choose_target_mds(MetaRequest *req, Inode** phash_diri)
 
   Inode *in = NULL;
   Dentry *de = NULL;
+  bool is_notification_op = (req->get_op() == CEPH_MDS_OP_ADD_KAFKA_TOPIC ||
+                        req->get_op() == CEPH_MDS_OP_REMOVE_KAFKA_TOPIC ||
+                        req->get_op() == CEPH_MDS_OP_ADD_UDP_ENDPOINT ||
+                        req->get_op() == CEPH_MDS_OP_REMOVE_UDP_ENDPOINT);
 
   if (req->resend_mds >= 0) {
     mds = req->resend_mds;
     ldout(cct, 10) << __func__ << " resend_mds specified as mds." << mds << dendl;
     goto out;
   }
 
-  if (cct->_conf->client_use_random_mds)
+  if (cct->_conf->client_use_random_mds || is_notification_op)
```

Comment:
Do we really need random behavior here? Wouldn't it be better to send to rank0 mds or the first up one?